cpuinfo.go

```diff
@@ -64,36 +63,26 @@ var (
 // CPUInfo returns information about current system CPUs.
 // See https://www.kernel.org/doc/Documentation/filesystems/proc.txt
 func (fs FS) CPUInfo() ([]CPUInfo, error) {
-	data, err := util.ReadFileNoStat(fs.proc.Path("cpuinfo"))
+	file, err := os.Open(fs.proc.Path("cpuinfo"))
 	if err != nil {
 		return nil, err
 	}
-	return parseCPUInfo(data)
+	defer file.Close()
+	return parseCPUInfo(file)
 }
 
-func parseCPUInfoX86(info []byte) ([]CPUInfo, error) {
-	scanner := bufio.NewScanner(bytes.NewReader(info))
+func parseCPUInfoX86(info io.Reader) ([]CPUInfo, error) {
```

Comment:
I think we decided to prefer reading everything in a buffer first. While this is unlikely to change mid reading, I think it's what we decided to to in general. But not sure, @SuperQ wdyt?

Reply:
We have been doing that previously, although I think this pattern is fine also.  The OS is probably buffering the file content anyway.  And I think `io.Reader` is better to use as a parameter for most `parseXXX` functions because it's a little more generic than a string or byte slice.

If we do want to buffer the entire file first, then it would probably be better to create a reader from the byte slice and then pass the reader to the `parse` function instead of passing the byte slice directly.  The byte slice parameter here might just be left over from me thinking I would need random access instead of line by line access to parse certain file formats.

Reply:
Ok sounds reasonable, then let's keep is a reader.